mongo_connector/connector.py

```diff
@@ -732,84 +741,76 @@ def apply_exclude_fields(option, cli_values):
-            dest_ns_set = cli_values['dest_ns_set'].split(',')
+        if cli_values["gridfs_set"]:
+            option.value["gridfs"] = cli_values["gridfs_set"].split(",")
+
+        if cli_values["dest_ns_set"]:
+            ns_set = option.value["include"]
+            dest_ns_set = cli_values["dest_ns_set"].split(",")
             if len(ns_set) != len(dest_ns_set):
                 raise errors.InvalidConfiguration(
                     "Destination namespace set should be the"
                     " same length as the origin namespace set.")
-            option.value['mapping'] = dict(zip(ns_set, dest_ns_set))
+            option.value["mapping"] = dict(zip(ns_set, dest_ns_set))
+
+        # Check for duplicates
+        if option.value["include"]:
+            validate_no_duplicates(option.value["include"],
+                                   "Include namespace set")
+        if option.value["exclude"]:
+            validate_no_duplicates(option.value["exclude"],
+                                   "Exclude namespace set")
+        if option.value["gridfs"]:
+            validate_no_duplicates(option.value["gridfs"],
+                                   "GridFS namespace set")
+        if option.value["mapping"]:
+            validate_no_duplicates(list(option.value["mapping"].values()),
+                                   "Destination namespace set")
 
-        ns_set = option.value['include']
-        if len(ns_set) != len(set(ns_set)):
-            raise errors.InvalidConfiguration(
-                "Namespace set should not contain any duplicates.")
+    def apply_namespaces(option, cli_values):
+        if (option.value['include'] or option.value['exclude'] or
+                option.value['mapping'] or option.value['gridfs']):
+            return apply_disjoint_namespace_options(option, cli_values)
+        else:
+            return apply_unified_namespace_options(option, cli_values)
 
-        ex_ns_set = option.value['exclude']
-        if len(ex_ns_set) != len(set(ex_ns_set)):
-            raise errors.InvalidConfiguration(
-                "Exclude namespace set should not contain any duplicates.")
+    def apply_unified_namespace_options(option, cli_values):
+        merge_namespaces_cli(option, cli_values)
+        namespace_options = copy.deepcopy(option.value)
+        ns_set = namespace_options.pop('include', None)
+        ex_ns_set = namespace_options.pop('exclude', None)
+        gridfs_set = namespace_options.pop('gridfs', None)
+        dest_mapping = namespace_options.pop('mapping', None)
+        option.value["namespace_options"] = namespace_options
 
-        # not allow to exist both 'include' and 'exclude'
-        if ns_set and ex_ns_set:
-            raise errors.InvalidConfiguration(
-                "Cannot use both namespace 'include' "
-                "(--namespace-set) and 'exclude' "
-                "(--exclude-namespace-set).")
+        validate_namespace_options(
```

Comment:
`validate_namespace_options` is called twice. Once here when parsing the configuration file and command line options and again in `Namespaces.__init__`.  In this context,  LOG lines are printed to stdout because logging is not yet configured.  In `Namespaces.__init__` the warnings will be printed to the log file. 

This makes sense because it make any potential configuration problems more visible and it lets us keep the existing command line and config tests in `test_config.py`.